### PR TITLE
Don't add ambiently declared modules to ATA's unresolvedModules list

### LINKED ISSUE #20402: Duplicate identifier errors when NodeJS project has local copy of @types/node + ATA enabled
1. Create new NodeJS Console App
2. `npm install @types/node` in the project folder
3. Add ` "typeAcquisition": {"enable": true} ` to the `tsconfig.json`
3. Trigger a program recompute

Result: Duplicate identifier errors for all the declarations in the node .d.ts

Fix is to tell ATA to not ask the typings installer to add types that are part of the implicit type references of the project

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes #20402

Builds on #20403; **only the last commit is new relative to that one**

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
N.B. from issue

> Fix is to tell ATA to not ask the typings installer to add types that are part of the implicit type references of the project

This turns out to be insufficient - even after excluding `node` (because it's part of the implicit type references of the project), we go add it again later in the typingsInstaller because we see e.g. `fs` in the `unresolvedImports` of one of the *local* `@types/node`'s .d.ts files, and helpfully map the `fs` module to `node`. Note that this mapping occurs in the typingsInstaller which doesn't even know that the `node` file is the types package for `node`.

The prior fix in #20403 is sufficient in cases where an ambient module is declared in the location where we *would* find it if it were a proper module, but doesn't fix the case where it's simply ambient in a random location. The simple additional fix here (which doesn't per se exclude the other one, because the other fix improves performance) is to ask for the list of ambient modules from the checker and exclude them.

File: src/compiler/program.ts [hunk 1]
```diff
@@ -825,9 +824,10 @@ namespace ts {
             // If we change our policy of rechecking failed lookups on each program create,
             // we should adjust the value returned here.
             function moduleNameResolvesToAmbientModuleInNonModifiedFile(moduleName: string, oldProgramState: OldProgramState): boolean {
-                const resolutionToFile = getResolvedModule(oldProgramState.file, moduleName);
-                if (resolutionToFile) {
-                    // module used to be resolved to file - ignore it
+                const resolutionToFile = getResolvedModule(oldProgramState.oldSourceFile, moduleName);
+                const resolvedFile = resolutionToFile && oldProgramState.program && oldProgramState.program.getSourceFile(resolutionToFile.resolvedFileName);
+                if (resolutionToFile && !resolvedFile.externalModuleIndicator) {
+                    // module used to be resolved to module - ignore it
```
Comment (sheetalkamat) [inline src/compiler/program.ts]: The comment here says that the module use to be resolved to module and yet we check !resolvedFile.externalModuleIndicator?

Comment (RyanCavanaugh) [reply src/compiler/program.ts]: This is for the case where we previously resolved to a module that ultimately contained an ambient declaration for the module we were looking for, but isn't actually itself the module. In that case we can't "re-use" the resolution.

File: src/server/project.ts [hunk 2]
```diff
@@ -506,6 +506,13 @@ namespace ts.server {
         }
         abstract getTypeAcquisition(): TypeAcquisition;
 
+        protected removeLocalTypingsFromTypeAcquisition(newTypeAcquisition: TypeAcquisition): TypeAcquisition {
+            if (!newTypeAcquisition || !newTypeAcquisition.include) {
```
Comment (billti) [inline src/server/project.ts]: Is this right? `if (!something) return something` always reads a bit fishy.

File: src/compiler/program.ts (hunk 1 above)
Comment (sheetalkamat) [reply src/compiler/program.ts]: That makes sense, can you please update the comment.

**Pull request merged**